### ISSUE TITLE
Not returning serviceworker.js

### ISSUE THREAD

**Issue by bloggerpact** [meanbee/magento-meanbee-pwa#18]
Hey @tgerulaitis 

I used the modman to install the plugin, everything looks okay from the back-end have https enabled cleared cache but the url.com/serviceworker.js service worker does not return Javascript. I'm using custom theme & also tried with rwd theme. 
In waiting for your quick response

**Comment by tgerulaitis** [comment]
Hi @bloggerpact,

What version of Magento are you running? Do you have "Allow Symlinks" set to "Yes" in _System > Configuration > Advanced > Developer > Template Settings_ (You might not have that setting anymore if you're running on the latest version of Magento)?

First thing I would try is running:
```
modman deploy --copy magento-meanbee-pwa
```
Then flushing the Magento cache and running through setup scripts.

**Comment by fintzen** [comment]
I cleared cache and become this error message.
**console.log('Service worker registration failed:', err);**
url.com/serviceworker.js is just a blank page - not code.

In Safari = Blank page
In Firefox = Blank page
In Chrome = /serviceworker.js Failed to load resource: net::ERR_INVALID_RESPONSE
(index):95 Service worker registration failed: TypeError: Failed to register a ServiceWorker: A bad HTTP response code (500) was received when fetching the script.

I use magento 1.9.1.1 and i have "Allow symlinks" enabled.
What do I'm wrong?

**Comment by tgerulaitis** [comment]
@fintzen, looks like there's an exception being thrown on the server, resulting in a 500 HTTP response code. Could you check `var/log/exception.log` and `var/log/system.log` for any relevant error messages?

**Comment by fintzen** [comment]
@tgerulaitis i found this 3 errors in system.log
-------------------------------------
2018-01-09T09:36:01+00:00 ERR (3): Warning: simplexml_load_string(): Entity: line 55: parser error : Start tag expected, '&lt;' not found  in /html/magento/app/code/core/Mage/Core/Model/Layout/Update.php on line 450
2018-01-09T09:36:01+00:00 ERR (3): Warning: simplexml_load_string():   in /html/magento/app/code/core/Mage/Core/Model/Layout/Update.php on line 450
2018-01-09T09:36:01+00:00 ERR (3): Warning: simplexml_load_string(): ^  in /html/magento/app/code/core/Mage/Core/Model/Layout/Update.php on line 450

**Comment by tgerulaitis** [comment]
@fintzen , that's strange. The error messages seem to indicate that you have an invalid layout xml file. I would check `app/design/frontend/base/default/layout` and `app/design/frontend/<yourpackage>/<yourtheme>/layout` for any xml files that have invalid code at the start. Although, I'm not sure if this is going to be related to your problems generating the service worker file.

Just to eliminate another common issue, what web server are you using for Magento? Apache or Nginx? If it's Nginx, does it allow `/serviceworker.js` to be served by PHP? (I've seen a lot of Nginx configurations that assume anything ending in `.js` is a file and not letting PHP respond)

**Comment by fintzen** [comment]
@tgerulaitis , now i found the invalid code at the start of a page.xml file. 
But it was not the problem that was related to the service worker problem.

I use a apache server for magento. 
I really have no idea why the service worker is not working.

**Comment by tgerulaitis** [comment]
@fintzen, Yeah, I'm not sure either, unfortunately. I've set up a fresh instance of Magento 1.9.1.1 myself locally to test this and I'm not seeing any errors running the module. Unfortunately, without more information about the problem there's nothing else I can do.

Have you looked at your web server error logs or PHP logs?

**Comment by fintzen** [comment]
I can't find any other error logs on my server.
Now I have a little bit trying to start the ServiceWorker. 
I put the .js on direct in the directory url.com/serviceworker.js
Now I have a other error message.
<img width="727" alt="serviceworker_request_failed" src="https://user-images.githubusercontent.com/16959522/34919918-a196485a-f96a-11e7-8071-4e72b59836b7.png">

**Comment by tgerulaitis** [comment]
@fintzen the `serviceworker.js` file in the `templates` directory is not a valid service worker file. It's a template that contains PHP tags which Magento should evaluate and replace with appropriate values. It's not going to work if being served directly through the web server.

**Comment by fintzen** [comment]
@tgerulaitis, oh yes I'm really sorry for that dumb comment. 😅 
I just see my server can't find the resource - so I put it in there to test if it's run. 
<img width="735" alt="bildschirmfoto 2018-01-15 um 11 46 56" src="https://user-images.githubusercontent.com/16959522/34938864-13863680-f9ea-11e7-9e0e-86597cd3c5d0.png">

**Comment by rafaelpatro** [comment]
Same problem with me!
Load layout was not loading properly. It was rendering a blank page too.
This was because another extension or theme was applying a block without a _referencing block_ (without ancestor, without parent). Only _root_ block could do it.

It was solved by locating the extension and updating [meanbee_pwa.xml](https://github.com/meanbee/magento-meanbee-pwa/blob/master/src/app/design/frontend/base/default/layout/meanbee_pwa.xml#L10).
Added a new removal tag: `<remove name="sucking-extension-block" />`

Good luck!

**Comment by fintzen** [comment]
@rafaelpatro, Wich extension / wich block was the problem? Maybe we use the same.

**Comment by rafaelpatro** [comment]
Extension: Mirasvit Searchautocomplete
File: _/app/design/frontend/base/default/layout/mst_searchautocomplete.xml_
```xml
<?xml version="1.0"?>
<layout version="0.1.0">
    <default>
        <reference name="head">
            <action method="addJs" ifconfig="searchautocomplete/general/enabled"><script>mirasvit/core/jquery.min.js</script></action>
            <action method="addJs" ifconfig="searchautocomplete/general/enabled"><script>mirasvit/core/underscore.js</script></action>
            <action method="addJs" ifconfig="searchautocomplete/general/enabled"><script>mirasvit/core/backbone.js</script></action>
            <action method="addJs" ifconfig="searchautocomplete/general/enabled"><script>mirasvit/code/searchautocomplete/form.js</script></action>
            <action method="addJs" ifconfig="searchautocomplete/general/enabled"><script>mirasvit/code/searchautocomplete/autocomplete.js</script></action>
            <action method="addItem" ifconfig="sales/msrp/enabled"><type>skin_js</type><name>js/msrp.js</name></action>
        </reference>

        <block type="searchautocomplete/layout" name="search.block">
            <action method="addSearchAutocomplete" ifconfig="searchautocomplete/general/enabled" />
        </block>
```

# Solution
Edit: /app/design/frontend/base/default/layout/meanbee_pwa.xml
Add a removal tag for _search.block_:

```xml
<?xml version="1.0"?>
<layout>
    <default>
        <reference name="after_body_start">
            <block name="serviceworker.register" type="meanbee_pwa/serviceworker_register" template="pwa/serviceworker/register.phtml"/>
        </reference>
    </default>

    <serviceworker_index_js>
        <remove name="root"/>
        <remove name="search.block"/>
        <block name="serviceworker.js" type="meanbee_pwa/serviceworker_js" template="pwa/serviceworker/serviceworker.js" output="toHtml"/>
    </serviceworker_index_js>
</layout>
```

**Comment by tgerulaitis** [comment]
Excellent, thank you @rafaelpatro !
